docs/sphinx/_ext/sphinx-mdx-builder/sphinxcontrib/mdxbuilder/builders/mdx.py

```diff
@@ -0,0 +1,76 @@
+from os import path
+from typing import Iterator
+
+from docutils import nodes
+from docutils.io import StringOutput
+
+from sphinx.builders import Builder
+from sphinx.util import logging
+from sphinx.util.osutil import ensuredir
+
+from ..writers.mdx import MdxWriter
+
+logger = logging.getLogger(__name__)
+
+
+class MdxBuilder(Builder):
+    name = "mdx"
+    format = "mdx"
+    epilog: str = "______ [Finished writing mdx files for %(project)s to %(outdir)s] ______"
+    allow_parallel = True
+    file_suffix: str = ".mdx"
+    link_suffix: str | None = None  # defaults to file_suffix
+
+    def init(self):
+        if self.config.mdx_file_suffix is not None:
+            self.file_suffix = self.config.mdx_file_suffix
+        if self.config.mdx_link_suffix is not None:
+            self.link_suffix = self.config.mdx_link_suffix
+        elif self.link_suffix is None:
+            self.link_suffix = self.file_suffix
+
+        def file_transform(docname: str) -> str:
+            return docname + self.file_suffix
+
+        def link_transform(docname: str) -> str:
+            return docname + (self.link_suffix or self.file_suffix)
+
+        self.file_transform = self.config.mdx_file_transform or file_transform
+        self.link_transform = self.config.mdx_link_transform or link_transform
+
+    def get_outdated_docs(self) -> Iterator[str]:
```

Comment:
This is neat, I'm surprised it's a required override and not built-in (referencing [this doc](https://www.sphinx-doc.org/en/master/extdev/builderapi.html#sphinx.builders.Builder.get_outdated_docs)).

So only updated source files will be re-rendered?